### PR TITLE
Add OTLP over gRPC project skeleton.

### PR DESCRIPTION
**Pull request by Yun-Ting**
## Changes
Working towards: https://github.com/open-telemetry/opentelemetry-dotnet/issues/3421#issuecomment-1314630702

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <PackageTags>$(PackageTags);OTLP;gRPC</PackageTags>
+    <MinVerTagPrefix>core-</MinVerTagPrefix>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
+    <PackageReference Include="Google.Protobuf" Version="$(GoogleProtobufPkgVer)" />
+    <PackageReference Include="Grpc.Tools" Version="$(GrpcToolsPkgVer)" PrivateAssets="all" />
+    <PackageReference Include="Grpc.Net.Client" Version="$(GrpcNetClientPkgVer)" />
+  </ItemGroup>
```
Comment (alanwest) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj:15]: nit: we usually put PackageReferences and ProjectReferences in their own ItemGroups... though, admittedly we have probably not kept completely consistent.

```suggestion
  <ItemGroup>
    <PackageReference Include="Google.Protobuf" Version="$(GoogleProtobufPkgVer)" />
    <PackageReference Include="Grpc.Tools" Version="$(GrpcToolsPkgVer)" PrivateAssets="all" />
    <PackageReference Include="Grpc.Net.Client" Version="$(GrpcNetClientPkgVer)" />
  </ItemGroup>
  
  <ItemGroup>
    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
  </ItemGroup>
```

**Review by alanwest — APPROVED** [review]
Looks like a fine place to start. @utpilla I can't recall if having this empty project in the repo will have any affect on the release process. Will it cause any pain? i.e., need to manually exclude it from the push to nuget or something?

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <Description>OpenTelemetry protocol exporter over gRPC for OpenTelemetry .NET</Description>
+    <PackageTags>$(PackageTags);OTLP;gRPC</PackageTags>
+    <MinVerTagPrefix>core-</MinVerTagPrefix>
+  </PropertyGroup>
+
```
Comment (alanwest) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj:9]: I think you need to add the following to fix the CI.

```suggestion

  <!--Do not run ApiCompat as this is newly added. Remove this property once we have released a stable version.-->
  <PropertyGroup>
    <RunApiCompat>false</RunApiCompat>
  </PropertyGroup>

```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# OTLP Exporter over gRPC for OpenTelemetry .NET
+
+[![NuGet](https://img.shields.io/nuget/v/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.svg)](https://www.nuget.org/packages/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc)
+[![NuGet](https://img.shields.io/nuget/dt/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.svg)](https://www.nuget.org/packages/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc)
+
+[The OTLP (OpenTelemetry Protocol) gRPC exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter)
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/README.md:6]: Broken link

Comment (utpilla) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/README.md:6]: "gRPC based implementation of OTLP Exporter" is probably a better description

**Comment by utpilla** [comment]
> Looks like a fine place to start. @utpilla I can't recall if having this empty project in the repo will have any affect on the release process. Will it cause any pain? i.e., need to manually exclude it from the push to nuget or something?

The publish to MyGet workflow runs `dotnet pack` on `OpenTelemetry.proj`. We need to do something similar to what we have for SemanticConventions: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/OpenTelemetry.proj#L7-L9

**Review by cijothomas — APPROVED** [review]
LGTM.
We need to handle the migration smoothly. Let existing pacakge's readme or changelog reflect the fact that therei s a new separate package, etc.

File: OpenTelemetry.proj [hunk 4]
```diff
@@ -7,6 +7,8 @@
     <!-- Not pack SemanticConventions project for now -->
     <SolutionProjects Remove="src\OpenTelemetry.SemanticConventions\OpenTelemetry.SemanticConventions.csproj" />
     <PackProjects Remove="src\OpenTelemetry.SemanticConventions\OpenTelemetry.SemanticConventions.csproj" Condition="'$(OS)' != 'Windows_NT'" />
+    <SolutionProjects Remove="src\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj" />
+    <PackProjects Remove="src\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj" />
```
Comment (CodeBlanch) [inline OpenTelemetry.proj:11]: @Yun-Ting Just curious, why is this needed?

Comment (CodeBlanch) [reply OpenTelemetry.proj:11]: Chatted with @utpilla about it offline.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <Description>OpenTelemetry protocol exporter over gRPC for OpenTelemetry .NET</Description>
+    <PackageTags>$(PackageTags);OTLP;gRPC</PackageTags>
+    <MinVerTagPrefix>core-</MinVerTagPrefix>
+  </PropertyGroup>
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj:8]: For new projects I would really like us to turn on things like...

```xml
    <Nullable>enable</Nullable>
    <AnalysisLevel>latest-all</AnalysisLevel>
    <ImplicitUsings>enable</ImplicitUsings>
```

...but we're probably going to be copying in a lot of stuff here from the existing project?

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Comment by Yun-Ting** [comment]
> LGTM. We need to handle the migration smoothly. Let existing pacakge's readme or changelog reflect the fact that therei s a new separate package, etc.

Thanks for the suggestion. I will do a follow-up PR to do the README.md update for the existing package.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj (hunk 5 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc/OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj:8]: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3924
